docs/src/main/sphinx/connector/iceberg.rst

```diff
@@ -377,12 +377,15 @@ above, this SQL will delete all partitions for which ``country`` is ``US``::
     DELETE FROM iceberg.testdb.customer_orders
     WHERE country = 'US'
 
-Currently, the Iceberg connector only supports deletion by partition.
-This SQL below will fail because the ``WHERE`` clause selects only some of the rows
-in the partition::
+Both v1 and v2 tables will perform a partition delete if the ``WHERE`` clause
+meets these conditions.
 
-    DELETE FROM iceberg.testdb.customer_orders
-    WHERE country = 'US' AND customer = 'Freds Foods'
+Row level deletion
+^^^^^^^^^^^^^^^^^^
+
+Tables using v2 of the Iceberg specification support deletion of individual rows
+using the ``merge-on-read`` mode. In order to use this feature the ``format_version``
+table property must be set to ``2``.
```

Comment:
- for new Tables, from Trino perspective, user doesn't need to set `format_version=2`, because it's the default
- for existing tables, referring to `format_version` may be confusing because we don't show it yet (https://github.com/trinodb/trino/pull/11980) and don't allow changing it (tbd)


let's just say the table must be in "v2" iceberg spec version